shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java

```diff
@@ -86,21 +122,102 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
+        } else {
+            addAttachmentButton.setVisibility(View.GONE);
+            attachmentImageView.setImageURI(screenshotUri);
+            attachmentImageView.setOnClickListener(createNavigationClickListener());
+        }
+    }
+
+    private void showDocumentPicker() {
+        Intent intent = new Intent(Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT
+                                   ? Intent.ACTION_OPEN_DOCUMENT
+                                   : Intent.ACTION_GET_CONTENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        intent.setType(ALL);
+        // only choosing from local storage due to permission issues
+        // and lack of loading progress when downloading from a network location
+        intent.putExtra(Intent.EXTRA_LOCAL_ONLY, true);
+        intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
+            intent.addFlags(Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION);
+        }
+        startActivityForResult(intent, ATTACHMENT_REQUEST_CODE);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        if (requestCode == ATTACHMENT_REQUEST_CODE) {
+            if (resultCode == Activity.RESULT_OK) {
+                List<Uri> list = new ArrayList<>();
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN
+                    && intent.getClipData() != null) {
+                    ClipData clipData = intent.getClipData();
+                    for (int i = 0, size = clipData.getItemCount(); i < size; i++) {
+                        list.add(clipData.getItemAt(i).getUri());
+                    }
+                } else if (intent.getData() != null) {
+                    list.add(intent.getData());
+                }
+                Context context = getContext();
+                for (Uri uri : list) {
+                    Utils.persistFilePermissions(context, uri, intent);
+                    addAttachment(uri, null, true);
+                }
+            }
+        } else {
+            super.onActivityResult(requestCode, resultCode, intent);
+        }
+    }
+
+    private void addAttachment(@Nullable Uri fileUri, @Nullable String defaultFilename, boolean removable) {
+        if (fileUri == null) {
+            return;
+        }
+        View view = LayoutInflater.from(getContext()).inflate(R.layout.shaky_attachment_view, attachmentsView, false);
+        AttachmentViewHolder viewHolder = new AttachmentViewHolder(view, onAttachmentClickListener);
+        viewHolder.bind(fileUri, defaultFilename, removable);
+        attachmentsView.addView(view);
+    }
+
+    private void removeAttachment(@NonNull Uri fileUri) {
+        for (int i = 0, count = attachmentsView.getChildCount(); i < count; i++) {
```

Comment:
We may want to consider using an adapter here, since this seems like it would lend itself to some kind of list or grid...

Reply:
I thought about that but in that case we need to use ListView  (for RecyclerView we need to add the dependencies). In most of the cases, users won't add too many attachments or remove them so I think it might be fine to add/remove views from a view group.
Maybe it is easier to use a ListView in this case, I can change it later.